src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -311,6 +328,33 @@ private Paging GetClientMethodPaging(Method method, Method nextPageMethod, IDict
             }
         }
 
+        private static LongRunningOperation BuildLongRunningOperation(Method method, Response originalResponse, IDictionary<object, object> options)
+        {
+            var originalResponseParameter = new Parameter(
+                "originalResponse",
+                "The original response from starting the operation.",
+                new CSharpType(typeof(AzureResponse)),
+                null,
+                true);
+            var httpMessageParameter = new Parameter(
+                "createOriginalHttpMessage",
+                "Creates the HTTP message used for the original request.",
+                new CSharpType(typeof(Func<>), new CSharpType(typeof(HttpMessage))),
```

Comment:
Does `new CSharpType(typeof(Func<HttpMessage>)` work?

Reply:
I thought we had to provide generic parameters as a parameter... At least that's how the original implementation worked. Is that no longer the case? Other places in the code use this pattern seen here.

Reply:
I'm not sure :) there is no reason to require it if all generic parameters are available as C# types.

Reply:
I double checked. Currently, we don't 'pull apart' the Type passed in. We just get Name, Namespace, and IsValueType out of it. I think that is because it would involve recursion in the constructor which I avoided when I originally made the first concept of this thing. For now, this works, but I created an issue to address allowing the shorthand creation: https://github.com/Azure/autorest.csharp/issues/458